selfservice/hook/web_hook.go

```diff
@@ -234,192 +218,186 @@ func (e *WebHook) ExecutePostRegistrationPostPersistHook(_ http.ResponseWriter,
-		return e.execute(ctx, &templateContext{
-			Flow:           flow,
-			RequestHeaders: req.Header,
-			RequestMethod:  req.Method,
-			RequestURL:     x.RequestURL(req).String(),
-			RequestCookies: cookies(req),
-			Identity:       id,
-		})
-	})
+	return e.execute(req.Context(), &templateContext{
+		Flow:           flow,
+		RequestHeaders: req.Header,
+		RequestMethod:  req.Method,
+		RequestURL:     x.RequestURL(req).String(),
+		RequestCookies: cookies(req),
+		Identity:       id,
+	}, "SettingsPostPersistHook")
 }
 
 func (e *WebHook) ExecuteSettingsPrePersistHook(_ http.ResponseWriter, req *http.Request, flow *settings.Flow, id *identity.Identity) error {
 	if !(gjson.GetBytes(e.conf, "can_interrupt").Bool() || gjson.GetBytes(e.conf, "response.parse").Bool()) {
 		return nil
 	}
-	return otelx.WithSpan(req.Context(), "selfservice.hook.WebHook.ExecuteSettingsPrePersistHook", func(ctx context.Context) error {
-		return e.execute(ctx, &templateContext{
-			Flow:           flow,
-			RequestHeaders: req.Header,
-			RequestMethod:  req.Method,
-			RequestURL:     x.RequestURL(req).String(),
-			RequestCookies: cookies(req),
-			Identity:       id,
-		})
-	})
+	return e.execute(req.Context(), &templateContext{
+		Flow:           flow,
+		RequestHeaders: req.Header,
+		RequestMethod:  req.Method,
+		RequestURL:     x.RequestURL(req).String(),
+		RequestCookies: cookies(req),
+		Identity:       id,
+	}, "SettingsPrePersistHook")
 }
 
-func (e *WebHook) execute(ctx context.Context, data *templateContext) error {
-	var (
-		httpClient     = e.deps.HTTPClient(ctx)
-		ignoreResponse = gjson.GetBytes(e.conf, "response.ignore").Bool()
-		canInterrupt   = gjson.GetBytes(e.conf, "can_interrupt").Bool()
-		parseResponse  = gjson.GetBytes(e.conf, "response.parse").Bool()
-		emitEvent      = gjson.GetBytes(e.conf, "emit_analytics_event").Bool() || !gjson.GetBytes(e.conf, "emit_analytics_event").Exists() // default true
-		tracer         = trace.SpanFromContext(ctx).TracerProvider().Tracer("kratos-webhooks")
-	)
-	if ignoreResponse && (parseResponse || canInterrupt) {
-		return errors.WithStack(herodot.ErrInternalServerError.WithReasonf("A webhook is configured to ignore the response but also to parse the response. This is not possible."))
-	}
-
-	makeRequest := func() (finalErr error) {
-		if ignoreResponse {
-			// This means we want to run this closure asynchronously and not be
-			// canceled when the parent context is canceled.
-			//
-			// The webhook will still cancel after 30 seconds as that is the
-			// configured timeout for the HTTP client.
-			ctx = context.WithoutCancel(ctx)
-		}
-		ctx, span := tracer.Start(ctx, "selfservice.webhook")
-		defer otelx.End(span, &finalErr)
-
-		if emitEvent {
-			instrumentHTTPClientForEvents(ctx, httpClient)
+func (e *WebHook) execute(ctx context.Context, data *templateContext, hookName string) error {
+	return otelx.WithSpan(ctx, "selfservice.hook.WebHook.Execute"+hookName, func(ctx context.Context) error {
```

Comment:
Sorry for bikeshedding, but please don't do this. Right now, the `selfservice.hook.WebHook.Execute SettingsPrePersistHook` literals are quickly greppable, which this change destroys.